master/custom/factories.py

```diff
@@ -169,6 +169,9 @@ class UnixNoGilBuild(UnixBuild):
     buildersuffix = ".nogil"
     configureFlags = ["--with-pydebug", "--disable-gil"]
     factory_tags = ["nogil"]
+    # 2024-04-11: Free-threading can still be slower than regular build in some
+    # code paths, so tolerate longer timeout.
+    test_timeout = int(TEST_TIMEOUT * 1.5)
 
 
 class UnixNoGilRefleakBuild(UnixBuild):
```

Comment:
Should we increase the timeout for this one too?

Reply:
UnixNoGilRefleakBuild already uses `TEST_TIMEOUT * 10` and I didn't see any test failing with this timeout. I prefer to not make it bigger (until a test fails with timeout), the timeout is already larger than 3 hours! (`worker timeout: 3 hour 25 min`)